motionplan/motionPlanner.go

```diff
@@ -511,6 +513,47 @@ func CheckPlan(
 		return err
 	}
 
+	// create a list of segments to iterate through
+	var segments []*ik.Segment
+	if relative {
+		segments = make([]*ik.Segment, 0, len(poses)+1)
+
+		// get the inputs we were partway through executing
+		checkFrameGoalInputs, ok := plan.Trajectory()[0][checkFrame.Name()]
+		if !ok {
+			return errors.New("could not get inputs for checkFrame")
+		}
+
+		// get checkFrame's currentInputs
+		checkFrameCurrentInputs, ok := currentInputs[checkFrame.Name()]
```

Comment:
Why are you doing this?  sf.mapToSlice() will give you the same info without potentially ommitting info that is necessary